lib/collection/property-base.js

```diff
@@ -167,6 +169,15 @@ _.assign(PropertyBase.prototype, /** @lends PropertyBase.prototype */ {
      */
     parent: function () {
         return this && this.__parent && this.__parent.__parent || undefined;
+    },
+
+    /**
+     * Accepts an object and sets it as the parent of the current property.
+     *
+     * @param {Object} parent The object to set as parent.
+     */
+    setParent: function (parent) {
+        _.assignLocked(this, __PARENT, parent);
```

Comment:
Only do this if no parent currently exists, because if one is already there, this will throw an error.

We're assigning the property with `writable: false`